kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigDependentResource.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.kroxylicious.kubernetes.operator;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.api.model.ConfigMapBuilder;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+import io.javaoperatorsdk.operator.api.reconciler.dependent.managed.ManagedWorkflowAndDependentResourceContext;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.CRUDKubernetesDependentResource;
+import io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependent;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+import io.kroxylicious.proxy.config.Configuration;
+
+import static io.kroxylicious.kubernetes.operator.Labels.standardLabels;
+import static io.kroxylicious.kubernetes.operator.ProxyConfigStateData.CONFIG_OBJECT_MAPPER;
+import static io.kroxylicious.kubernetes.operator.ResourcesUtil.namespace;
+
+/**
+ * Generates a Kube {@code ConfigMap} containing the proxy config YAML.
+ */
+@KubernetesDependent
+public class ProxyConfigDependentResource extends CRUDKubernetesDependentResource<ConfigMap, KafkaProxy> {
+
+    public static final String CONFIG_YAML_KEY = "proxy-config.yaml";
+    public static final String REASON_INVALID = "Invalid";
+
+    /**
+     * The key of the {@code config.yaml} entry in the desired {@code Secret}.
+     */
+
+    public static final String SECURE_VOLUME_KEY = "secure-volumes";
+    public static final String SECURE_VOLUME_MOUNT_KEY = "secure-volume-mounts";
+    public static final String CONFIGURATION_DATA_KEY = "configuration";
+
+    public ProxyConfigDependentResource() {
+        super(ConfigMap.class);
+    }
+
+    /**
+     * @return The {@code metadata.name} of the desired ConfigMap {@code Secret}.
+     */
+    static String configMapName(KafkaProxy primary) {
+        return ResourcesUtil.name(primary);
```

Comment:
with the introduction of the second configmap it's now possible to cause problems by having two proxies one named `simple` and `simple-config-state` as both will try to manifest a `simple-config-state` ConfigMap for different reasons.

I think we should add a suffix to this one too, something like `-proxy-config`